hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsConfigHelper.java

```diff
@@ -93,11 +93,23 @@ public static void overrideClientMetricsConfig(ClientConfig config, ILogger logg
         ClientMetricsConfig metricsConfig = config.getMetricsConfig();
         MetricsJmxConfig jmxConfig = metricsConfig.getJmxConfig();
 
+        //Check old deprecated STATISTICS settings first.
+        // MetricsConfig.collectionFrequencySeconds
+        tryOverride(ClientProperty.STATISTICS_PERIOD_SECONDS, config::getProperty,
```

Comment:
Is this consistent with other configs? I always thought that whatever is in `ClientConfig` always is the highest priority and overrides the system environment property.

Reply:
Yes, I believe. At least, I have copied the same code a couple of lines below to make sure that I am consistent within Metrics Config.

Other than Metrics Config, I could not come up with a config which has a property counterpart but 
In a recent version, we have made a change so that one can set any value from the property and override it. See
https://docs.hazelcast.com/hazelcast/5.0-beta-1/configuration/overriding-configuration-settings.html

If you accept it as a similar use-case, there anything set from property overrides xml and programmatic configuration.

Reply:
wow, this is a surprise to me. i always thought that it worked the other way.